python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+additional methods to calculate band magnitudes using the LSST software stack
+after applying MW extinction:
+ -  calc_mags which use the magnitude calculations in LSST stack
+ -  extinction which use the extinction calculations in LSST stack
+
+"""
+import numpy as np
+
+from lsst.sims.photUtils.Sed import Sed
+from lsst.sims.photUtils.EBV import EBVbase
+from lsst.sims.photUtils.BandpassDict import BandpassDict
+from lsst.sims.photUtils.SignalToNoise import calcSNR_m5, calcMagError_m5
+from lsst.sims.photUtils.PhotometricParameters import PhotometricParameters
+
+import sncosmo
+
+__all__ = ['SNObject']
+
+
+class SNObject(sncosmo.Model):
+
+    """
+    Extension of the SNCosmo `TimeSeriesModel` to include more parameters and
+    use methods in the catsim stack. We constrain ourselves to the use of a
+    specific SALT model for the Supernova (Salt2-Extended), and set its MW
+    extinction to be 0, since we will use the LSST software to calculate
+    extinction.
+
+    Parameters
+    ----------
+    ra : float
+         ra of the SN in degrees
+    dec : float
+        dec of the SN in degrees
+
+
+    Attributes
+    ----------
+    _ra : float or None
+        ra of the SN in radians
+
+    _dec : float or None
+        dec of the SN in radians
+
+    skycoord : `np.ndarray' of size 2 or None
+        np.array([[ra], [dec]]), which are in radians
+
+    ebvofMW : float or None
+        mwebv value calculated from the self.skycoord if not None, or set to
+        a value using self.set_MWebv. If neither of these are done, this value
+        will be None, leading to exceptions in extinction calculation.
+        Therefore, the value must be set explicitly to 0. to get unextincted
+        quantities.
+
+    Methods
+    -------
+
+    Examples
+    --------
+    >>> SNObject  = SNObject(ra=30., dec=60.)
+    >>> SNObject._ra
+    >>> 0.5235987755982988
+    >>> SNObject._dec
+    >>> 1.0471975511965976
+    """
+
+    def __init__(self, ra=None, dec=None, source='salt2-extended'):
+        """
+        Parameters
+        ----------
+        ra : float
+            ra of the SN in degrees
+        dec : float
+            dec of the SN in degrees
+        """
+
+        dust = sncosmo.OD94Dust()
+        sncosmo.Model.__init__(self, source=source, effects=[dust, dust],
+                               effect_names=['host', 'mw'],
+                               effect_frames=['rest', 'obs'])
+
+        # Current implementation of Model has a default value of mwebv = 0.
+        # ie. no extinction, but this is not part of the API, so should not
+        # depend on it, set explicitly in order to unextincted SED from
+        # SNCosmo. We will use catsim extinction from `lsst.sims.photUtils`.
+
+        self.ModelSource = source
+        self.set(mwebv=0.)
+
+        # ra and dec if passed are assumed to be in degrees and converted into
+        # radians.
+        self._ra = ra
```

Comment:
```I think you can get away with this because setCoords sets self._ra and self._dec to radians. Maybe you should just get rid of this assignment of self._ra and self._dec.
```
Yes, that was the plan. The reason I had something before that was to check if ra, dec were provided or were NoneTypes (default). This check could be done on the input parameters  ra,dec. 
But I wanted to provide values in SNstate too (which right now is taken directly from self._ra, self._dec.